developer-docs-site/docs/guides/sign-a-transaction.md

```diff
@@ -0,0 +1,300 @@
+* **sequence_number** (uint64): Sequence number of this transaction. This must match the sequence number stored in the sender's account at the time the transaction executes.
+* **payload**: Instructions for the Aptos Blockchain, including publishing a module, execute a script function or execute a script payload.
+* **max_gas_amount** (uint64): Maximum total gas to spend for this transaction. The account must have more than this gas or the transaction will be discarded during validation.
+* **gas_unit_price** (uint64): Price to be paid per gas unit. During execution the `total_gas_amount`, calculated as: `total_gas_amount = txn_base_cost * gas_unit_price`, must not exceed `max_gas_amount` or the transaction will abort during the execution. `txn_base_cost` represents the total units of gas consumed when executing the transaction.
+* **expiration_timestamp_secs** (uint64): The blockchain timestamp at which the blockchain would discard this transaction.
+* **chain_id** (uint8): The chain ID of the blockchain that this transaction is intended to be run on.
+
+### BCS
+
+Binary Canonical Serialization (BCS) is a serialization format. See [BCS][bcs] for a description of the design goals of BCS.
+
+:::note
+
+BCS does not support floats, unicode chars and sets.
+
+:::
+
+BCS is not a self-describing format. As such, in order to deserialize a message, one must know the message type and layout ahead of time.
+
+An example of how BCS serializes a string.
+```typescript
+// A string is serialized as: byte length + byte representation of the string. The byte length is required for deserialization. Without it, no way the deserializer knows how many bytes to deserialize.
+const bytes: Unint8Array = bcs_serialize_string("aptos");
+assert(bytes == [5, 0x61, 0x70, 0x74, 0x6F, 0x73]);
+```
+
+### Signing message
+
+A signing message is the bytes of the BCS serialized raw transaction. There is also a prefix, `prefix_bytes`, which is `sha3_256("APTOS::RawTransaction")`. Therefore:
+
+`signing_message = prefix_bytes + bcs_bytes_of_raw_transaction.`
+
+The below code shows how prefix bytes are calculated.
+
+```typescript
+function signinMessagePrefix(): Buffer {
+  let hash = SHA3.sha3_256.create();
+  hash.update(`APTOS::RawTransaction`);
+  return Buffer.from(hash.arrayBuffer());
+}
+```
+
+### Signature
+A signature is the [Ed25519][ed25519] encryption of a signing message with the client private key. The signature is mainly used for security purpose.
+
+* By signing a signing message with the private key, clients prove to the Aptos Blockchain that they have authorized the transaction be executed.
+* Aptos Blockchain will validate the signature with client account's public key to ensure that the transaction submitted is indeed signed by the client.
+
+### Signed transaction
+
+A signed transaction consisted of:
+
+- A raw transaction, and
+- An authenticator. The authenticator contains a client's public key and the signature of the raw transaction.
```

Comment:
feels like it is easier to start from the biggest pieces (SignedTransaction -> (Authenticator, RawTransaction) -> how to create a raw transaction and how to create an authenticator

Reply:
I still like the bottom-up approach since it fits the flow we highlighted in the overview. I feel what we have is natural and clear.